cli-messaging/src/test/java/com/salesforce/messaging/EventKeyTest.java

```diff
@@ -61,66 +58,61 @@ public void verifyKeyInJson() {
+    @MethodSource("getAllWarningEventKeyValues")
+	public void verifyWarning(EventKey eventKey) {
+		assertThat("Unexpected messageType on EventKey." + eventKey, eventKey.getMessageType(), is(MessageType.WARNING));
+		// No verbose check since we don't have a specific rule for verbosity on warning
 	}
 
-	@Test
-	public void verifyErrorExternal() {
-		if (!eventKey.getMessageKey().startsWith(ERROR_EXTERNAL)) {
-			return;
-		}
-		assertEquals("Unexpected messageType on EventKey." + eventKey, MessageType.ERROR, eventKey.getMessageType());
-		assertFalse("Verbose value on external error messages should be True. Please recheck EventKey." + eventKey, eventKey.isVerbose());
-		assertEquals("MessageHandler on external error messages should be UX. Please recheck EventKey." + eventKey, MessageHandler.UX, eventKey.getMessageHandler());
+    @ParameterizedTest(name = "eventKey={0}")
+    @MethodSource("getAllErrorExternalEventKeyValues")
+	public void verifyErrorExternal(EventKey eventKey) {
+		assertThat("Unexpected messageType on EventKey." + eventKey, eventKey.getMessageType(), is(MessageType.ERROR));
+		assertThat("Verbose value on external error messages should be False. Please recheck EventKey." + eventKey, eventKey.isVerbose(), is(false));
+		assertThat("MessageHandler on external error messages should be UX. Please recheck EventKey." + eventKey, eventKey.getMessageHandler(), is(MessageHandler.UX));
 	}
 
-	@Test
-	public void verifyErrorInternal() {
-		if (!eventKey.getMessageKey().startsWith(ERROR_INTERNAL)) {
-			return;
-		}
-		assertEquals("Unexpected messageType on EventKey." + eventKey, MessageType.ERROR, eventKey.getMessageType());
-		assertFalse("Verbose value on internal error messages should be True. Please recheck EventKey." + eventKey, eventKey.isVerbose());
-		assertEquals("MessageHandler on internal error messages should be INTERNAL. Please recheck EventKey." + eventKey, MessageHandler.INTERNAL, eventKey.getMessageHandler());
+    @ParameterizedTest(name = "eventKey={0}")
+    @MethodSource("getAllErrorInternalEventKeyValues")
+	public void verifyErrorInternal(EventKey eventKey) {
+		assertThat("Unexpected messageType on EventKey." + eventKey, eventKey.getMessageType(), is(MessageType.ERROR));
+		assertThat("Verbose value on internal error messages should be False. Please recheck EventKey." + eventKey, eventKey.isVerbose(), is(false));
+		assertThat("MessageHandler on internal error messages should be INTERNAL. Please recheck EventKey." + eventKey, eventKey.getMessageHandler(), is(MessageHandler.INTERNAL));
 	}
 
-	// Needed to make this test run dynamically for each EntryKey value
+    public static Stream<EventKey> getAllEventKeyValues() {
+        return Arrays.stream(EventKey.values());
+    }
 
-	private EventKey eventKey;
+    public static Stream<EventKey> getAllInfoEventKeyValues() {
+        return getAllEventKeyValues().filter(eventKey -> eventKey.getMessageType() == MessageType.INFO);
```

Comment:
Why do this method and `getAllWarningEventKeyValues()` filter by `getMessageType() ==` instead of `getMessageKey().startsWith()`? As is, since the tests themselves check `getMessageType()` they're basically just checking A == A.

Reply:
Notice that INFO_TELEMETRY starts with "info." and yet it isn't of MessageType.INFO ... it is MessageType.TELEMETRY. That's why I chose to use the MessageType wheenever possible instead of the naming convention. Thoughts?

Reply:
Fixed.